### PR TITLE
updates for pe v3.8.* and 2015.2

### PR DESCRIPTION
**Pull request by abuxton**

### CODE REVIEW

**Comment by WhatsARanjit** [comment]
@abuxton Not to dole out more work, but can you add the corresponding spec tests?

**Comment by WhatsARanjit** [comment]
@abuxton Also, this module does work on 3.7.x with the NC API.  The puppet_gem provider wasn't introduced yet.

**Comment by abuxton** [comment]
hey Ranjit, i've now added tests and squashed the commit as no one needs to see that mess :)
Not sure about your last comment, the class is not needed in 3.8 hence why it just is executed in noop mode if some one should classify it for any reason!  
#puppethack

File: spec/shared_contexts.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+shared_context :windows_hiera do
+    # example only,
+    let(:hiera_data) do
+        {:some_key => "some_value" }
+    end
+end
+
+shared_context :real_hiera_data do
+    let(:hiera_config) do
+       hirea_config_file
```
Comment (WhatsARanjit) [inline spec/shared_contexts.rb]: Typo here?

**Comment by abuxton** [comment]
Yeah, retrospec file not required as no hieradata in the module, so I have removed it.

**Pull request closed without merging**